### PR TITLE
Add scittle to runtime versions

### PR DESCRIPTION
**Pull request by benjamin-asdf**
**Replace this placeholder text with a summary of the changes in your PR.
The more detailed you are, the better.**

-----------------

Before submitting the PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [X] The commits are consistent with our [contribution guidelines](../blob/master/.github/CONTRIBUTING.md)
- [ ] You've added tests (if possible) to cover your change(s)
- [ ] All tests are passing (`eldev test`)
- [ ] All code passes the linter (`eldev lint`) which is based on [`elisp-lint`](https://github.com/gonewest818/elisp-lint) and includes
  - [byte-compilation](https://www.gnu.org/software/emacs/manual/html_node/elisp/Byte-Compilation.html), [`checkdoc`](https://www.gnu.org/software/emacs/manual/html_node/elisp/Tips.html), [check-declare](https://www.gnu.org/software/emacs/manual/html_node/elisp/Declaring-Functions.html), packaging metadata, indentation, and trailing whitespace checks.
- [X] You've updated the [changelog](../blob/master/CHANGELOG.md) (if adding/changing user-visible functionality)
- [ ] You've updated the [user manual](../blob/master/doc) (if adding/changing user-visible functionality)

### CODE REVIEW

**Comment by benjamin-asdf** [comment]
Scittle nrepl was saying a version but we don't take it into account for runtime classification.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 - [#3588](https://github.com/clojure-emacs/cider/issues/3588): Compatibility with pwsh 7.3 quoting rules.
 - Bump the injected `enrich-classpath` to [1.19.0](https://github.com/clojure-emacs/enrich-classpath/compare/v1.18.6...v1.19.0).
 - Bump the `parseedn` required version to 1.2.1.
+- [#3600](https://github.com/clojure-emacs/cider/pull/3600) Fix scittle jack in when using `cider-jack-in-clj`
```
Comment (bbatsov) [inline CHANGELOG.md:10]: You're missing a `:` and a `.` here.

File: cider-connection.el [hunk 2]
```diff
@@ -456,12 +457,19 @@ But helps us know if this is a nbb repl, or not."
     (when nrepl-versions
       (nrepl-dict-get nrepl-versions "nbb-nrepl"))))
 
+(defun cider--scittle-nrepl-version ()
+  "Likewise for scittle. See `cider--nbb-nrepl-version'."
```
Comment (bbatsov) [inline cider-connection.el:461]: I'd rather copy the docstring from `nbb`, so this can be read standalone.

**Review by vemv — COMMENTED** [review]
I'd appreciate understanding, does scittle have a nrepl server? Mind to share some links?

If so ideally it would bave the same capabilities as a bb-nrepl server, namely `ns-list` and `classpath` ops, for correct Sesman functioning.

**Comment by vemv** [comment]
You could also expand the user manual to show that this is possible, its use case, sample setup etc.

...It would also help us verify this works as intended - I've never used scittle but would gladly QA this branch

**Comment by bbatsov** [comment]
@benjamin-asdf ping :-)

**Comment by bob-just-bob** [comment]
https://github.com/babashka/scittle/tree/main/doc/nrepl

**Comment by benjamin-asdf** [comment]
Hi, the write up is here. 
https://github.com/babashka/scittle/tree/main/doc/nrepl

**Comment by vemv** [comment]
> But we removed this mainly because of the startup performance of doing a cljs eval or something?

Please check the git logs as I don't immediately recall what that thing was named, or why it was removed.

> The problem right now is that if somebody like Jack Rusher writes a new clojurescript repl, it won't work directly.

Not sure of how things came to that point (if they really are). Per:

```lisp
(defun cider-runtime ()
  "Return the runtime of the nREPl server."
  (cond
   ((cider--clojure-version) 'clojure)
   ((cider--babashka-version) 'babashka)
   ((cider--nbb-nrepl-version) 'nbb)
   ((cider--scittle-nrepl-version) 'scittle)
   (t 'generic)))
```

...the only cljs repls that need a special identification happen to come from the same author. There's a nice variety of cljs repls that don't need that.

**Comment by bbatsov** [comment]
@benjamin-asdf Ping :-)

**Comment by benjamin-asdf** [comment]
@bbatsov Jo I fixed the remaining lint error (I hope)

File: cider-connection.el [hunk 3]
```diff
@@ -457,12 +458,22 @@ But helps us know if this is a nbb repl, or not."
     (when nrepl-versions
       (nrepl-dict-get nrepl-versions "nbb-nrepl"))))
 
+(defun cider--scittle-nrepl-version ()
+  "Retrieve the underlying connection's scittle version."
+  (with-current-buffer
```
Comment (bbatsov) [inline cider-connection.el:463]: I'd use here more compact indentation, as the code reads better this way - e.g. like in the function above for `nbb`.

**Comment by bbatsov** [comment]
I think you'll need to rebase on top of `master`, as the changelog seems outdated on this branch. I also left a few small remarks.

**Comment by benjamin-asdf** [comment]
@bbatsov Cool, I fixed the remaining issues

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**